src/Raven.Server/Documents/ETL/EtlTransformer.cs

```diff
@@ -164,6 +171,40 @@ private JsValue LoadCounter(JsValue self, JsValue[] args)
             return loadCounterReference;
         }
 
+        private JsValue LoadTimeSeries(JsValue self, JsValue[] args)
+        {
+            const int paramsCount = Transformation.TimeSeriesTransformation.LoadTimeSeries.ParamsCount;
+            const string signature = Transformation.TimeSeriesTransformation.LoadTimeSeries.Signature;
+            
+            if (args.Length != paramsCount)
+                ThrowInvalidScriptMethodCall($"{signature} must have {paramsCount} arguments");
+                
+            if(args[0].IsString() == false)
+                ThrowInvalidScriptMethodCall($"{signature}. The argument timeSeriesName must be string");
+            var timeSeriesName = args[0].AsString();
+
+            var from = ScriptRunner.GetDateArg(args[1], signature, "from"); 
+            var to = ScriptRunner.GetDateArg(args[2], signature, "to"); 
+                
+            JsValue loadTimeSeriesReference = (JsValue)Transformation.TimeSeriesTransformation.Marker + timeSeriesName;
+
+            if ((Current.Document.Flags & DocumentFlags.HasTimeSeries) == DocumentFlags.HasTimeSeries)
+            {
+                var reader = Database.DocumentsStorage.TimeSeriesStorage.GetReader(Context, Current.DocumentId, timeSeriesName, from, to);
+                //TODO Maybe to return null if no entries
```

Comment:
*can* this return null? I thought it would return empty list?

Reply:
If the time-series does not exist so `LoadTimeSeries` will return `null` (not in the current implementation).
I thought to return `null` also if the time-series does exist but there are no entries in the range.